heudiconv/tests/test_convert.py

```diff
@@ -111,3 +114,75 @@ def test_b0dwi_for_fmap(tmpdir, capfd):
         assert op.isdir(op.join(tmppath, 'sub-%s', mod) % (subID))
         for ext in ['bval', 'bvec']:
             assert glob(op.join(tmppath, 'sub-%s', mod, 'sub-%s_*.%s') % (subID, subID, ext))
+
+
+# Test two scenarios for each case:
+# -study without sessions
+# -study with sessions
+@pytest.mark.parametrize(
+    "subjects, sesID, expected_session_folder", [
+        (['Jason', 'Bourne'], None, 'sub-{sID}'),
+        (['Bourne'], 'Treadstone', op.join('sub-{{sID}}', 'ses-{{ses}}')),
+    ]
+)
+# Two possibilities: with or without heuristics:
+@pytest.mark.parametrize(
+    "heuristic", ['example', None]       # heuristics/example.py
+)
+def test_convert(tmpdir, monkeypatch, capfd,
```

Comment:
```suggestion
def test_populate_intended_for(tmpdir, monkeypatch, capfd,
```
since the entire file is to test convert in various scenarios - test name should be more specific